python/sagemaker-multumodel-endpoint/README.md

```diff
@@ -0,0 +1,91 @@
+* [Amazon SageMaker](https://aws.amazon.com/sagemaker/) - Build, train, and deploy machine learning (ML) models for any use case with fully managed infrastructure, tools, and workflows
+* [Docker](https://www.docker.com/) – Docker is a set of platform as a service products that use OS-level virtualization to deliver software in packages called containers.
+* Based on the [Sagemaker Notebook from Amazon Sagemaker Examples](https://github.com/aws/amazon-sagemaker-examples/blob/master/advanced_functionality/multi_model_bring_your_own/multi_model_endpoint_bring_your_own.ipynb)
+
+## Project structure
+
+    ├── cdk_sagemaker_hosting_multiple_models   # Directory containing the AWS CDK stacks describing the infrastructure. See https://docs.aws.amazon.com/cdk/latest/guide/stacks.html
+    ├── models                                  # Directory containing the models that will deployed by the CDK stack
+    ├── docker                                  # Directory containing files for model Docker deployment
+    ├── examples                                # Directory containing files used for retrieving example models and testing deployment
+    └── images                                  # Directory containing the architecture diagram
+
+## Prerequisites
+- [Create AWS Account](https://aws.amazon.com/premiumsupport/knowledge-center/create-and-activate-aws-account/).
+- [Install the AWS CDK Toolkit](https://docs.aws.amazon.com/cdk/v2/guide/cli.html) (`cdk` CLI command), e.g. by running:
+```bash
+npm install -g aws-cdk
+```
+- [Install Docker](https://www.docker.com/get-started). *Ensure the Docker daemon is running when executing deployment*
+- [Bootstrap the AWS account and region](https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html) to provision resources required for deploying CDK stacks:
+```bash
+cdk bootstrap
+```
+----
+
+## Deploy an example
+
+1. Create a Python virtual environment
+```bash
+python3 -m venv .venv
+```
+2. Activate your Python virtual environment
+```bash
+source .venv/bin/activate
+```
+Successful activation can be verified by running `pip -V`, the output should show that pip is being accessed at a path that contains the `.venv` directory.
+3. Install the required dependencies
+```bash
+pip install -r requirements.txt
+```
+4. Navigate to the `examples` directory and execute `download_models.py`
+5. Navigate to the root directory and execute `cdk deploy`. After successful deployment our endpoint will be running two models
+6. To validate successful deployment, navigate to the `examples` directory and execute `invoke_endpoint.py` (*Ensure you are using the same AWS credentials used when deploying*). The first invocation of a model may be slow, since behind the scenes, SageMaker is downloading the model artifacts from S3 to the instance and loading it into the container.
+
```

Comment:
```suggestion
4. Navigate to the `examples` directory and execute `download_models.py`
	  ```bash
	  cd examples
	  python3 -m download_models.py
	   ```
5. Navigate to the root directory and execute `cdk deploy`. After successful deployment our endpoint will be running two models
	  ```bash
	  cd ..
	  cdk deploy
	   ```
6. To validate successful deployment, navigate to the `examples` directory and execute `invoke_endpoint.py` (*Ensure you are using the same AWS credentials used when deploying*). The first invocation of a model may be slow, since behind the scenes, SageMaker is downloading the model artifacts from S3 to the instance and loading it into the container.
	  ```bash
	  cd examples
	  python3 -m invoke_endpoint
	   ```
```

Let's document the CLI commands in these steps too.

Reply:
One minor improvement could probably be made here. Couldn't we run all these commands without stepping in and out of the examples folder? For example we should be able to just do `python3 -m examples.download_models` from the root dir. The paths in the code would need to be updated to accommodate this however it seems like a nice dev UX enhancement.  Not a huge deal if you don't have time to implement this now, I just want to know if i'm overlooking something here. We can always add it in as an enhancement later.